src/plugins/validation/2and3/semantic-validators/parameters-ibm.js

```diff
@@ -25,8 +25,99 @@ module.exports.validate = function({ jsSpec, isOAS3 }, config) {
+                  Array.isArray(
+                    obj.paths[head].get.responses[prop].content[mediaTypeKey]
+                      .schema[param]
+                  )
+                ) {
+                  topLevelIsArray = true;
+                  break;
+                }
+              }
+            }
+          }
+        }
+        if (obj.paths[head].get && topLevelIsArray) {
+          if (
+            obj.paths[head].get.parameters[0].limit &&
+            (typeof obj.paths['/pets'].get.parameters[0].limit !== 'number' ||
+              obj.paths[head].get.parameters[0].required.includes('limit'))
+          ) {
+            const message =
+              'limit parameter is optional and must be of type integer and must have a default value';
+            const checkStatus = config.invalid_limit_type;
+            if (checkStatus !== 'off') {
+              result[checkStatus].push({
+                path,
+                message
+              });
+            }
+          }
+          if (
+            obj.paths[head].get.parameters[0].start &&
+            (typeof obj.paths[head].get.parameters[0].start !== 'string' ||
+              obj.paths[head].get.parameters[0].required == true)
+          ) {
+            const message =
+              'start parameter must be of type string and must be optional';
+            const checkStatus = config.invalid_limit_type;
+            if (checkStatus !== 'off') {
+              result[checkStatus].push({
+                path,
+                message
+              });
+            }
+          }
+          if (
+            obj.paths[head].get.parameters[0].cursor &&
+            (typeof obj.paths[head].get.parameters[0].cursor !== 'string' ||
+              obj.paths[head].get.parameters[0].required == true)
+          ) {
+            const message =
+              'cursor parameter must be of type string and must be optional';
+            const checkStatus = config.invalid_limit_type;
+            if (checkStatus !== 'off') {
+              result[checkStatus].push({
+                path,
+                message
+              });
+            }
+          }
+          if (
+            (!obj.paths['/pets'].get.parameters[0].start ||
+              !obj.paths['/pets'].get.parameters[0].cursor) &&
+            obj.paths['/pets'].get.parameters[0].offset &&
+            (typeof obj.paths['/pets'].get.parameters[0].offset !== 'number' ||
+              obj.paths['/pets'].get.parameters[0].required.includes(
+                'offset'
+              ) == true)
+          ) {
+            const message =
+              'if start or cursor parameters are not present then the offset parameter should be defined as an integer and should be optional';
+            const checkStatus = config.invalid_limit_type;
+            if (checkStatus !== 'off') {
+              result[checkStatus].push({
+                path,
+                message
+              });
+            }
+          }
```

Comment:
I don't see the checks for properties in the response that are specified in the issue.  These need to be added.